api/v1alpha1/shared_types.go

```diff
@@ -181,6 +181,13 @@ type KubernetesPodSpec struct {
 	//
 	// +optional
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+
+	// SchedulerName is the kubernetes scheduler that should be used for this pod.
+	// If specified, the pod will be dispatched by specified scheduler.
+	// If not specified, the pod will be dispatched by default scheduler.
+	//
+	// +optional
+	SchedulerName string `json:"schedulerName,omitempty"`
```

Comment:
if it's optional, make it a pointer.

Reply:
This matches the core [PodSpec API](https://pkg.go.dev/k8s.io/api/core/v1#PodSpec) and an empty string would already be processed as such. I can still change it if you'd like but aligning with core seemed preferable.

Reply:
let's keep same as other optional fileds in this repo.

Reply:
One problem with using a pointer is then we need a nil check or to set a default. For fields like SecurityContext, the K8s PodSpec type already passes through as a pointer and is handled there but for SchedulerName it wants a string.

Reply:
Updated as a pointer

Reply:
Should be resolved now @zirain